// @nteract/transform-vega/lib/react.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const React = __importStar(require("react"));
const external_1 = require("./external");
/** React component embedding a certain Vega (Lite) media type. */
class VegaEmbed extends React.Component {
    constructor(props) {
        super(props);
        this.anchorRef = React.createRef();
    }
    render() {
        return React.createElement("div", { ref: this.anchorRef });
    }
    async callEmbedder() {
        if (this.anchorRef.current === null) {
            return;
        }
        try {
            const result = await external_1.embed(this.anchorRef.current, this.props.mediaType, this.props.spec, this.props.options);
            if (this.props.resultHandler) {
                this.props.resultHandler(result);
            }
        }
        catch (error) {
            (this.props.errorHandler || console.error)(error);
        }
    }
    shouldComponentUpdate(nextProps) {
        return this.props.spec !== nextProps.spec;
    }
    componentDidMount() {
        this.callEmbedder().then();
    }
    componentDidUpdate() {
        this.callEmbedder().then();
    }
}
exports.VegaEmbed = VegaEmbed;
